spot_driver_cpp/src/spot_image_sources.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/spot_image_sources.hpp>
+#include <spot_driver_cpp/types.hpp>
+#include <tl_expected/expected.hpp>
+
+namespace
+{
+const std::vector<std::string> kImageSourceNames = {
+    "back",
+    "frontleft",
+    "frontright",
+    "left",
+    "right",
+};
+
+constexpr auto kImageSourceHand = "hand";
+
+constexpr auto kSuffixBodyColor = "_fisheye_image";
+constexpr auto kSuffixHandColor = "_color_image";
+
+constexpr auto kSuffixDepth = "_depth";
+constexpr auto kSuffixDepthRegistered = "_depth_registered";
+
+tl::expected<std::string, std::string> getBeforeSuffix(const std::string& string, const std::string& suffix)
+{
+    const auto index = string.find(suffix);
+    if (index != std::string::npos)
+    {
+      return string.substr(0, index);
+    }
+
+    return tl::make_unexpected("Did not find substring `" + suffix + "` within string `" + string + "`.");
+}
+}
+
+namespace spot_ros2
+{
+std::string toRosTopic(const ImageSource& image_source)
+{
+    if (image_source.type == SpotImageType::RGB)
+    {
+        return std::string("camera").append("/").append(image_source.name);
+    }
+    else if (image_source.type == SpotImageType::DEPTH)
+    {
+        return std::string("depth").append("/").append(image_source.name);
+    }
+    else  // SpotImageType::DEPTH_REGISTERED
```

Comment:
I think the only benefit from doing that would be if Spot gained support for an additional built-in camera, which would require updating the enums and conversion functions anyway. Seems unnecessary in my opinion, especially since I wouldn't actually be able to write a test case to trigger the default failure case given that we handle all possible values of each enums.